src/lambda_vm/lambda_vm.mli

```diff
@@ -1,38 +1,110 @@
-module Ast = Ast
-module Gas : module type of Gas
+module Ast : sig
+  type ident = string
 
-exception Out_of_stack
-exception Out_of_gas
+  type prim =
+    | Neg
+    | Add
+    | Sub
+    | Mul
+    | Div
+    | Rem
+    | Land
+    | Lor
+    | Lxor
+    | Lsl
+    | Lsr
+    | Asr
+    | Fst
+    | Snd
+
+  type expr =
+    (* calculus *)
+    | Var   of ident
+    | Lam   of ident * expr
+    | App   of {
+        funct : expr;
+        arg : expr;
+      }
+    (* prims *)
+    | Const of int64
+    | Prim  of prim
+    (* branching *)
+    | If    of {
+        predicate : expr;
+        consequent : expr;
+        alternative : expr;
+      }
+    (* memory *)
+    | Pair  of {
+        first : expr;
+        second : expr;
+      }
+
+  type value =
+    | Int64 of int64
+    | Pair  of value * value
+
+  type script = {
+    param : ident;
+    code : expr;
+  }
+end
+module Gas : sig
+  type t
+
+  val make : initial_gas:int -> t
+  val is_empty : t -> bool
+
+  val burn_constant : t -> unit
+  val burn_log2 : t -> cardinality:int -> unit
+end
+module Error_set : sig
+  type 'error t =
+    [ `Internal_error of 'error
+    | `Runtime_error  of [`Out_of_gas | `Out_of_stack] ]
+  [@@deriving show]
+
+  val pp_runtime_error :
+    Format.formatter -> [`Out_of_gas | `Out_of_stack] -> unit
+end
 
 (* ir *)
-type script
-type value
+module Ir : sig
```

Comment:
Not sure about adding the modules here, but if you add it, then remove the comments as they become redundant.